### PR TITLE
CCAP-112 Map children in need of CCAP

### PR DESCRIPTION
**Pull request by enyia21**
#### ✅ Completion tasks

- [ ] Added relevant tests
- [ ] Meets acceptance criteria

### CODE REVIEW

File: src/test/java/org/ilgcc/app/pdf/NeedChildcareForChildrenPreparerTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import formflow.library.data.Submission;
+import formflow.library.pdf.SingleField;
+import formflow.library.pdf.SubmissionField;
+import java.util.Map;
+import org.ilgcc.app.utils.SubmissionTestBuilder;
+import org.junit.jupiter.api.Test;
+
+public class NeedChildcareForChildrenPreparerTest {
+    NeedChildcareForChildren preparer = new NeedChildcareForChildren();
+    void setup(){
```
Comment (analoo) [inline src/test/java/org/ilgcc/app/pdf/NeedChildcareForChildrenPreparerTest.java:14]: do you need this?

File: src/test/java/org/ilgcc/app/utils/SubmissionTestBuilder.java [hunk 2]
```diff
@@ -339,4 +340,36 @@ public SubmissionTestBuilder regularPartnerScheduleWithCommuteTime(String commut
...
+        child.put("childAttendsOtherEd", (childAttendsOtherEd));
+        return this;
+    }
+    public SubmissionTestBuilder withChildIsAUSCitizen(int childIterationPosition, String childIsUsCitizen){
+        List<Map<String, Object>> children = (List<Map<String, Object>>) submission.getInputData().getOrDefault("children", emptyList());
+        if (children.isEmpty()) {
+            return this;
+        }
+
+        Map<String, Object> child = children.get(childIterationPosition);
```
Comment (analoo) [inline src/test/java/org/ilgcc/app/utils/SubmissionTestBuilder.java:360]: The childIsUSCitizen and childHasSpecial needs fields are not filled out in the filled out pdf. 

Could you update the test to ensure that this is the case?

Screenshot 2024-06-21 at 2 50 44 PM

**Review by analoo — CHANGES_REQUESTED** [review]

File: src/test/java/org/ilgcc/app/pdf/NeedChildcareForChildrenPreparerTest.java (hunk 1 above)
Comment (enyia21) [reply src/test/java/org/ilgcc/app/pdf/NeedChildcareForChildrenPreparerTest.java:14]: No need for setup. 
I'm removing it

Comment (enyia21) [reply src/test/java/org/ilgcc/app/pdf/NeedChildcareForChildrenPreparerTest.java]: Fixed

**Review by analoo — APPROVED** [review]
LGTM.

**Pull request merged**